facturacion_mexico/facturacion_fiscal/doctype/factura_fiscal_mexico/factura_fiscal_mexico.py

```
@@ -300,20 +314,66 @@ def _is_sales_invoice_return(self) -> bool:
 		sales_invoice = frappe.get_doc("Sales Invoice", self.sales_invoice)
 		return bool(getattr(sales_invoice, "is_return", 0))
 
+	def _is_nota_descuento(self) -> bool:
+		"""Determinar si la nota de crédito es por descuento/bonificación (no devolución física).
+
+		La intención de negocio se captura en el campo fm_tipo_nota_credito de la FFM.
+		Vacío o 'Devolución de mercancía' → False (devolución física, comportamiento actual).
+		"""
+		return (self.get("fm_tipo_nota_credito") or "").strip() == NOTA_CREDITO_DESCUENTO
+
+	def _detect_nota_credito_motivo(self):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_detect_nota_credito_motivo`

Add return type annotation: `None`

(ANN202)